share/cmake/Modules/FindSDL2.cmake

```diff
@@ -0,0 +1,18 @@
+# FPHSA_NAME_MISMATCHED is require to suppress warning message 
+set(FPHSA_NAME_MISMATCHED TRUE)
+include(FindPkgConfig)
+pkg_check_modules(sdl2 REQUIRED sdl2)
+unset(FPHSA_NAME_MISMATCHED)
+
+add_library(SDL2::SDL2 INTERFACE IMPORTED)
+set(SDL2_INCLUDE_DIRS ${sdl2_INCLUDE_DIRS})
+set(SDL2_LIBRARIES ${sdl2_LIBRARIES})
+set(SDL2_LINK_LIBRARIES ${sdl2_LINK_LIBRARIES})
+set_target_properties(SDL2::SDL2 PROPERTIES
+  INTERFACE_INCLUDE_DIRECTORIES "${SDL2_INCLUDE_DIRS}"
+  INTERFACE_LINK_LIBRARIES "${SDL2_LINK_LIBRARIES}"
+  # NOTE: pkg_check_modules' Cflags definition for "XBOX" did not get included and was passed to CFLAGS_OTHER for some reason... 
```

Comment:
nit: Most comments in this PR have typos (`pkg_check_modules'`) or poor wording. Someone should spell check these.

Reply:
It is not a typo, you can find it at https://cmake.org/cmake/help/latest/module/FindPkgConfig.html#command:pkg_check_modules . Since the function's name is already plural, then there shouldn't be an s after apostrophe?